### ISSUE TITLE
RPC from C++ doesn't work

### ISSUE THREAD

**Issue by Shatur** [godotengine/godot#50803]
### Godot version

4.0 (d779b5aa3ee7c2e93304f6c085675f8d3ec7aca5, first commit that breaks RPC)

### System information

Arch Linux, GLES3, Nvidia MX150

### Issue description

I trying to update https://github.com/godotengine/godot/pull/37200 to the latest master. After d779b5aa3ee7c2e93304f6c085675f8d3ec7aca5 RPC that called from C++ stopped working (works from GDScript). The previous commit works as expected. I think that this is a bug in the engine itself.

### Steps to reproduce

Previous code worked as expected:

```cpp
// In constructor
rpc_config("_rpc_server_send_inputs", MultiplayerAPI::RPC_MODE_REMOTE);

// Later call
node->rpc_unreliable_id(1, "_rpc_server_send_inputs", packet_data);
```

After d779b5aa3ee7c2e93304f6c085675f8d3ec7aca5 I updated the code to the following:

```cpp
// In constructor
rpc_config("_rpc_server_send_inputs", MultiplayerAPI::RPC_MODE_REMOTE, NetworkedMultiplayerPeer::TRANSFER_MODE_UNRELIABLE);

// Later call
node->rpc_id(1, "_rpc_server_send_inputs", packet_data);
```

And the rpc stopped working. I have the following error in the editor:

```
E 0:00:05:0064   _process_rpc: Condition "config.name == StringName()" is true.
  <C++ Source>   core/io/multiplayer_api.cpp:374 @ _process_rpc()
```

### Minimal reproduction project

Here is my PR with changes: https://github.com/AndreaCatania/godot/pull/28. It updates the feature to the latest master. But I bisected, the issue happens first after d779b5aa3ee7c2e93304f6c085675f8d3ec7aca5.
A reproduction project can be taken from this PR: https://github.com/GodotNetworking/example-project/pull/2 and should be used with https://github.com/godotengine/godot/pull/37200

**Comment by Faless** [comment]
It's a lot of code to digest, but you don't seem to initialize your `StringName`s:
https://github.com/Shatur/godot/blob/player_proto/modules/network_synchronizer/networked_controller.cpp#L49

I guess it should be:
```cpp
StringName NetworkedController::sn_rpc_server_send_inputs = SNAME("_rpc_server_send_inputs");
```

**Comment by Shatur** [comment]
> I guess it should be:

But `SNAME` just creates a function with a static member to allow caching. But In this case I do not need to cache it.
Anyway I tried it with `SNAME` and also tried calling it directly by name - no matter, the problem still exists.

Also the `StringName` changes was done after d779b5aa3ee7c2e93304f6c085675f8d3ec7aca5.

**Comment by Faless** [comment]
Oh yeah, should be:
```
StringName NetworkedController::sn_rpc_server_send_inputs = StringName("_rpc_server_send_inputs");
```

or:
```
rpc_config(SNAME("_rpc_server_send_inputs"), MultiplayerAPI::RPC_MODE_REMOTE, MultiplayerPeer::TRANSFER_MODE_UNRELIABLE);
```

What I'm saying is that I don't see in the code where you actually assign that `StringName` the string value of `"_rpc_server_send_inputs"`

**Comment by Shatur** [comment]
> Oh yeah, should be:

Yep!

> What I'm saying is that I don't see in the code where you actually assign that StringName 

It assigned [here](https://github.com/Shatur/godot/blob/player_proto/modules/network_synchronizer/register_types.cpp#L43). I think it was workaround when `StringNames` crashed the editor on exit when they were in static memory. One of your suggested options from the above should be used. Probably it will be more convenient to use `SNAME`?

**Comment by Shatur** [comment]
> Probably it will be more convenient to use SNAME

Just pushed the change. But I double checked - the rpc issue is still present.

**Comment by Faless** [comment]
@Shatur there should be a fix in  #50991 if you want to try it out.

**Comment by Shatur** [comment]
@Faless thank you, if fixes the issue!
